src/MySqlConnector/Serialization/MySqlSession.cs

```diff
@@ -84,6 +84,8 @@ public void DoCancel(MySqlCommand commandToCancel, MySqlCommand killCommand)
 				// blocking the other thread for an extended duration.
 				killCommand.CommandTimeout = 3;
 				killCommand.ExecuteNonQuery();
+
+				commandToCancel.IsCanceled = true;
```

Comment:
I want to handle this situation in `MySqlSession.FinishQuerying`; I can push that change.